File: Platforms/Common/Reign.Video.OpenGL/States/RasterizerState.cs
using Reign.Core;

namespace Reign.Video.OpenGL
{
	public class RasterizerState : DisposableResource, IRasterizerState
	{
		#region Properties
		private RasterizerStateDesc desc;
		#endregion

		#region Constructors
		public RasterizerState(IDisposableResource parent, IRasterizerStateDesc desc)
		: base(parent)
		{
			this.desc = (RasterizerStateDesc)desc;
		}
		#endregion

		#region Methods
		public void Enable()
		{
			#if !iOS && !ANDROID && !NaCl && !RPI
			GL.PolygonMode(GL.FRONT_AND_BACK, desc.fillMode);
			#endif
			
			if (desc.cullMode != GL.NONE)
			{
				GL.Enable(GL.CULL_FACE);
				GL.CullFace(desc.cullMode);
			}
			else
			{
				GL.Disable(GL.CULL_FACE);
			}

			#if !iOS && !ANDROID && !NaCl && !RPI
			if (desc.multisampleEnable)
			{
				GL.Enable(GL.MULTISAMPLE);
			}
			else
			{
				GL.Disable(GL.MULTISAMPLE);
			}
			#endif

			#if DEBUG
			Video.checkForError();
			#endif
		}
		#endregion
	}
}

File: Platforms/Win32/Reign_Video_D3D9_Component/RasterizerState.cpp
#include "pch.h"
#include "RasterizerState.h"

namespace Reign_Video_D3D9_Component
{
	#pragma region Constructor
	RasterizerStateCom::RasterizerStateCom(VideoCom^ video, RasterizerStateDescCom^ desc)
	{
		this->video = video;
		this->desc = desc;
	}
	#pragma endregion
	
	#pragma region Methods
	void RasterizerStateCom::Enable()
	{
		video->device->SetRenderState(D3DRS_FILLMODE, desc->fillMode);
		video->device->SetRenderState(D3DRS_CULLMODE, desc->cullMode);
		video->device->SetRenderState(D3DRS_MULTISAMPLEANTIALIAS, desc->multisampleEnable);
	}
	#pragma endregion
}